File: divide.m
function result = divide_real_numbers(num1, num2)
    if isnumeric(num1) && isnumeric(num2)
        if num2 ~= 0
            result = num1 / num2;
        else
            result = "Error: Cannot divide by zero.";
        end
    else
        result = "Error: Both inputs must be real numbers.";
    end
end

% Take input from the user
number1 = input('Enter the first number: ', 's');
number2 = input('Enter the second number: ', 's');

if ~isnan(str2double(number1)) && ~isnan(str2double(number2))
    result = divide_real_numbers(str2double(number1), str2double(number2));
    disp(['Result of division: ', num2str(result)]);
else
    disp('Error: Please enter valid real numbers.');
end

File: divide.py
def divide_real_numbers(num1, num2):
    if isinstance(num1, (int, float)) and isinstance(num2, (int, float)):
        if num2 != 0:
            return num1 / num2
        else:
            return "Error: Cannot divide by zero."
    else:
        return "Error: Both inputs must be real numbers."

# Take input from the user
number1 = input("Enter the first number: ")
number2 = input("Enter the second number: ")

if number1.replace('.', '', 1).isdigit() and number2.replace('.', '', 1).isdigit():
    result = divide_real_numbers(float(number1), float(number2))
    print("Result of division:", result)
else:
    print("Error: Please enter valid real numbers.")
